test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,357 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests.Scale
+{
+    public class ScaleHostEndToEndTests
+    {
+        private const string Function1Name = "Function1";
+        private const string Function2Name = "Function2";
+        private const string Function3Name = "Function3";
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabled)
+        {
+            JArray functionMetadata = new JArray
+            {
+                new JObject
+                {
+                    { "type", "testExtensionATrigger" },
+                    { "functionName", Function1Name }
+                },
+                new JObject
+                {
+                    { "type", "testExtensionATrigger" },
+                    { "functionName", Function2Name }
+                },
+                new JObject
+                {
+                    { "type", "testExtensionBTrigger" },
+                    { "functionName", Function3Name }
+                }
+            };
+
+            string hostJson =
+            @"{
+              ""extensions"": {
+                ""testExtensionA"" : {
+                  ""foo"": 16
+                },
+                ""testExtensionB"" : {
+                  ""bar"": 8
+                }
+              }
+            }";
+
+            string hostId = "test-host";
+            var loggerProvider = new TestLoggerProvider();
+
+            IHostBuilder hostBuilder = new HostBuilder();
+            hostBuilder.ConfigureLogging(configure =>
+            {
+                configure.SetMinimumLevel(LogLevel.Debug);
+                configure.AddProvider(loggerProvider);
+            });
+            hostBuilder.ConfigureAppConfiguration((hostBuilderContext, config) =>
+            {
+                // Adding host.json here
+                config.AddJsonStream(new MemoryStream(Encoding.ASCII.GetBytes(hostJson)));
+
+                // Adding app settings
+                config.AddInMemoryCollection(new List<KeyValuePair<string, string>>()
+                {
+                    new KeyValuePair<string, string>("app_setting1", "value1"),
+                    new KeyValuePair<string, string>("app_setting2", "value2")
+                });
+            })
+            .ConfigureWebJobsScale<TestConcurrencyStatusRepository, TestScaleMetricsRepository>(options =>
+            {
+                // configure scale options
+                options.IsTargetScalingEnabled = tbsEnabled;
+                options.CheckTargetScalerEnabled = ts =>
+                {
+                    return true;
+                };
+                options.MetricsPurgeEnabled = false;
+                options.FunctionMetadata = functionMetadata;
+            }, hostId)
+            .ConfigureTestExtensionAScale()
```

Comment:
I am thinking about "dynamic assembly loading", in this case we do not know about exact name of the method.
we can add a new interface:
```
    interface IConfigureExtensionScale
    {
        void ConfigureExtensionScale(IHostBuilder builder);
    }
```

this approach is similar to [IWebJobsStartup](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Hosting/IWebJobsStartup.cs) and [IWebJobsStartup2](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Hosting/IWebJobsStartup2.cs).